### ISSUE TITLE
Extend the example set with more messaging patterns

### ISSUE THREAD

**Issue by goodboy** [codypiersall/pynng#67]
I know @codypiersall and I have talked in passing about this before but I figured I'd make an issue (to keep me accountable) and to point to some resources I've sniffed out.

There also seems to be some reported issues (#39, #43) that can add to and/or benefit from this.

- There is a baseline set of examples from the original [nanomessage docs](https://nanomsg.org/gettingstarted/index.html/) as well as the [equivalent for `nng`](https://nanomsg.org/gettingstarted/nng/) (all written in C of course) that could be written out in Python.
- The other more lengthy task would be attempting to go through a variety of the more advanced patterns in the [zeromq guide](http://zguide.zeromq.org/page:all) and writing the equivalents in `pynng`.

This task is something I will likely need to accomplish in some part anyway as part of building out more advanced data streaming infrastructure for `piker` and as part of a set of more advanced features for `tractor` after an initial dev release.

**Comment by codypiersall** [comment]
Hi @goodboy, thanks for the issue.  Always a pleasure hearing from you!

More examples would *certainly* be helpful.  The zmq guide you linked is zmq's killer feature in my opinion.

**Comment by goodboy** [comment]
> The zmq guide you linked is zmq's killer feature in my opinion.

Agreed :100: and I think getting a decent bulk of those going with `trio` + `pynng` could be fuel for some great distributed systems design :smile_cat:.

I actually will likely add a more advanced example using `pynng` to `tractor`'s docs eventually to show off what can be done :surfer:

**Comment by MauiJerry** [comment]
I added some example code in my recent response #39 - a pair of pub/sub programs that use either the normal or the Trio Async methods.  My rPi project needed async on both the client and server ends.  It uses pub/sub to distribute data from server, and Pair1(polyamorous=True) to send commands from clients.

**Comment by geronimo-iia** [comment]
hello, adding also few example using curio in https://github.com/codypiersall/pynng/pull/76

**Comment by geronimo-iia** [comment]
zeromq guide is really a killer guide !
 I don't known if I'll had enough time, but I had the zyre protocol in mind, and it could be nice to see how it could be implemented with pynng.
